src/OrchardCore.Modules/OrchardCore.Setup/Startup.cs

```diff
@@ -1,20 +1,60 @@
 using System;
+using System.Linq;
 using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 using OrchardCore.Modules;
 
 namespace OrchardCore.Setup
 {
     public class Startup : StartupBase
     {
+        public Startup(IHostingEnvironment env)
```

Comment:
Didn't try the whole branch but just tried it by only copy paste the startup.

It works by just injecting an `IShellConfiguration configuration`.